grakn/rpc/session.py

```diff
@@ -47,7 +50,10 @@ def __init__(self, client, database: str, session_type: SessionType, options=Gra
 
         self._session_id = self._grpc_stub.session_open(open_req).session_id
         self._is_open = True
-        self._pulse = self._scheduler.enter(5, 1, self._transmit_pulse, ())
+        self._scheduler.enter(5, 1, self._transmit_pulse, ())
+        thread = Thread(target=self._scheduler.run)
+        # TODO: We should probably kill this thread when the session closes as it prevents the process closing.
+        thread.start()
```

Comment:
@alexjpwalker given we only use `thread` in this place, we could inline it like so:

```suggestion
        Thread(target=self._scheduler.run).start()
```